### PR TITLE
[Notebook] Links, Restricted Notebook Links, Search links

### LINKED ISSUE #6202: [Notebook] Links are sometimes not formatted
#### Summary

Sometimes when pasting in links you can get into a state where urls aren't formatted/purged of tags. Needs more investigation and a fix.

#### Expected vs Current Behavior

Should always work anytime an entry is edited in any way.

#### Steps to Reproduce

1. Copy and paste a whitelisted url (domain ending in nasa.gov) link in
2. To exit click rapidly in a blank area underneath the entries, notice the text is not formatted

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #6202 

### Describe your changes:

Significant simplification to highlight component. This will allow us to highlight links in notebook search as well. Also fixed a bug where locked notebooks weren't formatting links correctly.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/utils/textHighlight/TextHighlight.vue [hunk 1]
```diff
@@ -58,68 +47,11 @@ export default {
...
+            let regex = new RegExp(`(?<!<[^>]*)(${this.highlight})`, 'gi');
 
-            for (let i = 0; i < textSegments.length; i++) {
-                if (textSegments[i].toLowerCase() === this.highlight.toLowerCase()) {
-                    this.addHighlightSegment(textSegments[i]);
-                } else {
-                    this.addTextSegment(textSegments[i]);
-                }
-            }
+            return this.text.replace(regex, `<span class="${this.highlightClass}">${this.highlight}</span>`);
```
Comment (shefalijoshi) [inline src/utils/textHighlight/TextHighlight.vue:54]: Nice!

File: src/plugins/notebook/components/NotebookEntry.vue [hunk 2]
```diff
@@ -248,28 +250,15 @@ export default {
         },
         formattedText() {
             // remove ANY tags
-            let text = sanitizeHtml(this.entry.text, SANITIZATION_SCHEMA);
+            const text = sanitizeHtml(this.entry.text, SANITIZATION_SCHEMA);
 
-            if (this.editMode || !this.urlWhitelist) {
+            if (this.editMode || !this.urlWhitelist.length === 0) {
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEntry.vue:255]: haha, nope... leftover

File: src/plugins/notebook/components/NotebookEntry.vue [hunk 3]
```diff
@@ -355,6 +344,22 @@ export default {
         deleteEntry() {
             this.$emit('deleteEntry', this.entry.id);
         },
+        formattValidUrls(text) {
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEntry.vue:347]: I can always count on you @ozyx :)

**Review by shefalijoshi — APPROVED** [review]

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
Small changes, but working ok.

File: src/plugins/notebook/components/NotebookEntry.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEntry.vue]: fixed

File: src/plugins/notebook/components/NotebookEntry.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/notebook/components/NotebookEntry.vue]: got it!

**Review by ozyx — APPROVED** [review]
LGTM

**Review by shefalijoshi — APPROVED** [review]

**Pull request merged**